cirq/sim/density_matrix_simulator.py

```diff
@@ -528,7 +536,7 @@ def __init__(self, params: study.ParamResolver,
                          final_simulator_state=final_simulator_state)
         size = np.prod(protocols.qid_shape(self), dtype=int)
         self.final_density_matrix = np.reshape(
-            final_simulator_state.density_matrix, (size, size))
+            final_simulator_state.density_matrix, (size, size)).copy()
```

Comment:
Nit: I would switch the order of copy and reshape to reduce the chance of reshape also making a copy.

Reply:
done